plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/logcat/LogcatClassVisitor.kt

```diff
@@ -0,0 +1,55 @@
+package io.sentry.android.gradle.instrumentation.logcat
+
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.commons.AdviceAdapter
+
+class LogcatClassVisitor(
+    apiVersion: Int,
+    originalVisitor: ClassVisitor,
+    private val minLevel: LogcatLevel
+) :
+    ClassVisitor(apiVersion, originalVisitor) {
+    override fun visitMethod(
+        access: Int,
+        name: String?,
+        descriptor: String?,
+        signature: String?,
+        exceptions: Array<out String>?
+    ): MethodVisitor {
+        val mv = super.visitMethod(access, name, descriptor, signature, exceptions)
+        return LogMethodVisitor(mv, access, name, descriptor, minLevel)
+    }
+}
+
+class LogMethodVisitor(
+    mv: MethodVisitor?,
+    access: Int,
+    name: String?,
+    desc: String?,
+    private val minLevel: LogcatLevel
+) :
+    AdviceAdapter(Opcodes.ASM7, mv, access, name, desc) {
```

Comment:
we should also use here the same apiVersion as in the ClassVisitor instead of `Opcodes.ASM7`, so have to pass it as a parameter here

```suggestion
    AdviceAdapter(apiVersion, mv, access, name, desc) {
```